nuxeo-core-binarymanager-s3/src/main/java/org/nuxeo/ecm/core/storage/sql/S3BinaryManager.java

```diff
@@ -209,7 +209,8 @@ protected void abortOldUploads() throws IOException {
     protected void setupCloudClient() throws IOException {
         // Get settings from the configuration
         bucketName = getProperty(BUCKET_NAME_PROPERTY);
-        bucketNamePrefix = MoreObjects.firstNonNull(getProperty(BUCKET_PREFIX_PROPERTY), StringUtils.EMPTY);
+        // as bucket prefix is optional we don't want to use the fallback mechanism
+        bucketNamePrefix = MoreObjects.firstNonNull(properties.get(BUCKET_PREFIX_PROPERTY), StringUtils.EMPTY);
```

Comment:
Could you take this opportunity to remove usage of `MoreObjects` and just do `StringUtils.defaultString(properties.get(BUCKET_PREFIX_PROPERTY))`